src/apps/relay/mainrelay.h

```diff
@@ -176,6 +176,8 @@ enum _NET_ENG_VERSION {
 
 typedef enum _NET_ENG_VERSION NET_ENG_VERSION;
 
+typedef enum _DRAINMODE_STATE { DRAINMODE_NOT_ENALBED = 0, DRAINMODE_REQUESTED = 1, DRAINMODE_ENABLED } DRAINMODE_STATE;
```

Comment:
I'm not sure I follow why an enum is needed here.

We're either in drain mode or not, right?

A stare transition from "not draining" to "requested" to "draining" doesn't seem to be useful.

Reply:
The current process is staged.  

Step 1:  DRAINMODE_REQUESTED is flagged for the run_listener_server method to pick up
Step 2:  run_listener_server loop see's REQUESTED flag, then we turn on is_draining flag in each turn server.  Then drain mode transitions to DRAINMODE_ENABLED.

If I can get rid of the 2 staged approach, I won't need an enum here.  :)